### ISSUE TITLE
Workflow for importing videos AND swapping videos is seriously broken

### ISSUE THREAD

**Issue by walt93** [WWBN/AVideo#4852]
**Describe the bug**
Import a video from Bitchute. End up with a black video that doesn't play. Import that video myself from Bitchute, upload, swap video files - notice that I get the same result. Take video file, transcode it in handbrake and manually upload. Go to swap  the video files and now swap is not working for that video file.

The video that I uploaded, got excited about, went and spammed twenty places on social media is pointing to a dead file.

No matter what I do, I cannot upload a good video and swap it back into that slot.

Now I have twenty posts on social media that look like a fucking idiot because I can't fix the original video.

**To Reproduce**
1. Go to Embed a video link
2. Import this link https://www.bitchute.com/video/a8ApmHkWi2Ht/
3. Observe that the video doesn't play
4. Use an app like Total Video Downloader to download the video from https://www.bitchute.com/video/a8ApmHkWi2Ht/
5. Upload it again
6. Swap videos with the one in step 1
7. Observe that neither video plays
8. Transcode the video yourself using Handbrake
9. Upload it
10. Try, try, try again to swap it with the original file
11. Smash your forehead into the monitor, it just doesn't work

**Expected behavior**
Step 2 to work
Step 10 to work

**Comment by JoshWho** [comment]
Mine works with Bitchute May just be a ip ban or something


https://user-images.githubusercontent.com/39392449/113392578-f9125500-935a-11eb-8654-3324a1be375f.mp4

**Comment by DanielnetoDotCom** [comment]
@walt93 your link return me this...

![image](https://user-images.githubusercontent.com/18009293/113439198-745a2200-93c0-11eb-9ae3-bfd777256494.png)

It is forbidden on bitchute side

**Comment by DanielnetoDotCom** [comment]
Nevermind, bitchute was not implemented if you were not using the correct embed link.

with this update, we are able to convert regular bitchute links into an embed

**Comment by walt93** [comment]
I'm not tracking you as to what the issue is. Let me try to explain better - there are two problems here

1) Embed a video failed from the link given above

2) When I downloaded the mp4 through the bitchute link, and dropped it on the encoder, that encoded video failed to play.

3) The "Swap a video" function seems to have a weird bug where after the first swap, it will not swap the video again - 
  upload A 
  notice that A is bad 
  upload B
  swap A/B 
  delete B 
  then notice that your new A is bad
  upload C
   swap A/C - nothing
  video A is now unswappable

**Comment by DanielnetoDotCom** [comment]
1 and 2 test it on demo site.

3. I am aware if you swap videos while the encoder still transferring you will have problems.

**Comment by walt93** [comment]
it is quite possible that I did that, #3, not sure. I was quite in a rush as I had posted to social media and noticed that people were clicking on a dead video.

I tried on the demo site, the configuration is different - it's playing in an iframe. 

For my use case I don't want iframes - we're grabbing videos from other sites before the censors pull them down, so embedded is no good at all, at all...

Point being: the encoder configuration has problems with ... whatever file format is in -this- .mp4 - I suggest that checking this particular data may prove fruitful, as I've found others on Bitchute specifically that choke in this same fashion: https://www.bitchute.com/video/a8ApmHkWi2Ht/
